### PR TITLE
executor: fix a bug when do multi-update.

### LINKED ISSUE #3604: multi-update bug when alias one table to multi names
```sql
create table b (x int, y int);
insert into b values(1,2), (3,4), (5,6);
update b as m, b as n set m.x = m.x + 10;
select * from b;
```
on MySQL we'll get:
```
+------+------+
| x    | y    |
+------+------+
|   11 |    2 |
|   13 |    4 |
|   15 |    6 |
+------+------+
```
but on TiDB, we'll get:
```
+------+------+
| x    | y    |
+------+------+
|   11 |    2 |
|    3 |    4 |
|    5 |    6 |
+------+------+
```

### PR DESCRIPTION
**Pull request by winoros**
fix #3604 
PTAL @hanfei1991 @lamxTyler @hicqu @zimulala

### CODE REVIEW

File: executor/write_test.go [hunk 1]
```diff
@@ -543,6 +543,17 @@ func (s *testSuite) TestMultipleTableUpdate(c *C) {
 
 	r = tk.MustQuery("select * from t1")
 	r.Check(testkit.Rows("10", "10"))
+
+	// fix https://github.com/pingcap/tidb/issues/3604
+	testSQL = `drop table if exists t;
```
Comment (hanfei1991) [inline executor/write_test.go]: Execute them one by one ...

File: executor/write.go [hunk 2]
```diff
@@ -1110,7 +1110,8 @@ type UpdateExec struct {
 	OrderedList []*expression.Assignment
 
 	// updatedRowKeys is a map for unique (Table, handle) pair.
-	updatedRowKeys map[table.Table]map[int64]struct{}
+	// According to issue#3604, TableName is better than table.Table.
```
Comment (tiancaiamao) [inline executor/write.go]: Of course. Interface is not good choice as map key.
I just wonder that is the table name case-insensitive?

Comment (winoros) [inline executor/write.go]: No, all alphabet in table name will always be lower case.

**Comment by XuHuaiyu** [comment]
LGTM

File: executor/write.go (hunk 2 above)
Comment (shenli) [inline executor/write.go]: How about tables with the same name but in different database?

**Comment by winoros** [comment]
Closed this pr becuase this solution didn't solve the core issue.

**Pull request closed without merging**